WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsInterface.java

```diff
@@ -939,11 +940,13 @@ public void run() {
     }
 
     protected void notifyFetchErrorOnUiThread(final Exception error) {
-        if (mActivity == null || mActivity.isFinishing() || mListener == null) {
+        if (mContext == null
+                || (mContext instanceof Activity && ((Activity) mContext).isFinishing())
```

Comment:
Do we really need to check `isFinishing()` here? That seems like something the caller should deal with.

Reply:
I was only adapting the existing code without changing the logic and yeah, that check is awkward. I removed it with 0247fd6, letting the listener decide whether the Activity (if any) needs to be checked for finishing. It seems that almost all listeners are already checking for `isAdded` so, no new checks are needed.